### PR TITLE
Refactor some Str2html code

### PR DESCRIPTION
**Pull request by wxiaoguang**
This PR touches the most interesting part of the "template refactoring".

Before, "Tr", "Escape" and "Safe" are quite straightforward, they just escapes and casts.

But "Str2html" is special: it sanitizes the input and casts the "safe" result to HTML type.

During the refactoring, I can see that there are many legacy problems, for example:

1. Unclear variable type. Especially for "web/feed/convert.go": sometimes it uses text, sometimes it uses HTML.
2. Assign text content to "RenderedContent" field, for example: `  project.RenderedContent = project.Description` in web/org/projects.go
3. Assign rendered content to text field, for example: `r.Note = rendered content` in web/repo/release.go 
4. (possible) Incorrectly calling `{{Str2html .PackageDescriptor.Metadata.ReleaseNotes}}` in package/content/nuget.tmpl, I guess the name Str2html misleads developers to use it to "render string to html", but it only sanitizes .... well, if ReleaseNotes really contains HTML, then this is not a problem, but I haven't figured out it yet.

----

**There is still a question**: should we encourage to use "Str2html" in templates in the future? Actually, variables like "RenderedContent" are generated by "markup.Render", it has already been sanitized by the renderer. Although double-sanitizing doesn't cause side-effect, I don't know whether people would to keep using it in templates (for a sense of security) or just only use it when it is necessary (for making code to have clearly defined behaviors)

### CODE REVIEW

**Comment by silverwind** [comment]
Ideally we should ensure that Go's contextual auto-escaping works whereever possible to avoid needing such helpers. Are the limitations of this auto-escaper well-known?

**Comment by wxiaoguang** [comment]
> Ideally we should ensure that Go's [contextual auto-escaping](https://pkg.go.dev/html/template#hdr-Contexts) works wherever possible to avoid needing such helpers. Are the limitations of this auto-escaper well-known?

What I am doing is this. You can see we don't need "Safe" or "Escape" in templates anymore.

**Comment by 6543** [comment]
> should we encourage to use "Str2html" in templates in the future?

in my opinion no ... it's to error prone

**Review by lunny — APPROVED** [review]

File: routers/web/org/projects.go [hunk 1]
```diff
@@ -104,7 +105,7 @@ func Projects(ctx *context.Context) {
 	}
 
 	for _, project := range projects {
-		project.RenderedContent = project.Description
+		project.RenderedContent = templates.Str2html(project.Description) // FIXME: is it right? why not render?
```
Comment (6543) [inline routers/web/org/projects.go:108]: well it would be nice to know why not ...

Comment (wxiaoguang) [reply routers/web/org/projects.go:108]: But it is not in this PR's scope. It is just the old logic.

**Review by 6543 — APPROVED** [review]

**Pull request merged**